libcalico-go/lib/names/policy.go

```diff
@@ -102,6 +102,24 @@ func validateBackendTieredPolicyName(policy, tier string) error {
 	return nil
 }
 
+// ValidateTieredPolicyName validates v3 policy name, policies in the default tier can be named without the default. prefix.
+// Policy names in non default tier have to be in a format of tier.name
+func ValidateTieredPolicyName(policy, tier string) error {
+	if policyNameIsFormatted(policy) {
+		return nil
+	}
+
+	tier = TierOrDefault(tier)
+	parts := strings.SplitN(policy, ".", 2)
+	if len(parts) == 1 && tier == "default" {
+		return nil
+	}
+	if len(parts) != 2 || !strings.HasPrefix(policy, tier+".") {
```

Comment:
I am not sure this is correct?

It's possible to create a policy named something like this I think:

```
name: tier.some.name.with.dots
```

That would fail this `len(parts)` check wouldn't it?